test/008b-rating.test.js

```diff
@@ -0,0 +1,168 @@
+      .send({ rateValue: '2' });
+  });
+
+  describe('Rate Article', () => {
+    it('should not rate an article if user is not logged in',
+      async () => {
+        const response = await chai.request(server)
+          .patch(`${baseUrl}/rating/silicon-valley-bugsburney`)
+          .send({ rateValue: 5 });
+        expect(response.status).to.equal(401);
+        expect(response.body.error).to.equal('No token provided');
+      });
+
+    it('should not rate an article if user is not a mentor',
+      async () => {
+        const response = await chai.request(server)
+          .patch(`${baseUrl}/rating/silicon-valley-bugsburney`)
+          .set('Authorization', nonMentorToken)
+          .send({ rateValue: 5 });
+        expect(response.status).to.equal(401);
+        expect(response.body.error).to.equal('Only mentors can rate articles');
+      });
+
+    it('should not rate an article if value is not a number',
+      async () => {
+        const response = await chai.request(server)
+          .patch(`${baseUrl}/rating/silicon-valley-bugsburney`)
+          .set('Authorization', isMentorToken)
+          .send({ rateValue: 'a' });
+        expect(response.status).to.equal(400);
+        expect(response.body.error).to.equal('Value must be a number');
+      });
+
+    it('should not rate an article if value is less than 1 or greater than 5',
+      async () => {
+        const response = await chai.request(server)
+          .patch(`${baseUrl}/rating/silicon-valley-bugsburney`)
+          .set('Authorization', isMentorToken)
+          .send({ rateValue: '8' });
+        const errorMessage = 'Value must not be less than 1 or greater than 5';
+        expect(response.status).to.equal(400);
+        expect(response.body.error).to.equal(errorMessage);
+      });
+
+    it('should not rate an article if no value is provided',
+      async () => {
+        const response = await chai.request(server)
+          .patch(`${baseUrl}/rating/silicon-valley-bugsburney`)
+          .set('Authorization', isMentorToken)
+          .send({ rateValue: '' });
+        const errorMessage = 'Rate value must be provided';
+        expect(response.status).to.equal(400);
+        expect(response.body.error).to.equal(errorMessage);
+      });
+
+    it('should rate an article if value is valid and user is a mentor',
+      async () => {
+        const response = await chai.request(server)
+          .patch(`${baseUrl}/rating/silicon-valley-bugsburney`)
+          .set('Authorization', isMentorToken)
+          .send({ rateValue: '4' });
+        expect(response.status).to.equal(200);
+        expect(response.body.yourRating).to.equal('4');
+      });
+
+    it('should rate an article even if it already has a rating',
+      async () => {
+        const response = await chai.request(server)
+          .patch(`${baseUrl}/rating/tales-of-two-strangers-daffyduck`)
+          .set('Authorization', isMentorToken)
+          .send({ rateValue: '4' });
```

Comment:
Could you do a rating of say `5` or `3` here? That will really show the result of the computation. (4 + 4) / 2 seems too easy.